integration-tests/emulated/tests/bridges/bridge-hub-kusama/src/tests/snowbridge.rs

```diff
@@ -0,0 +1,681 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+use crate::*;
+use asset_hub_kusama_runtime::xcm_config::bridging::to_ethereum::BridgeHubEthereumBaseFee;
+use bridge_hub_kusama_runtime::{EthereumBeaconClient, EthereumInboundQueue, RuntimeOrigin};
+use codec::{Decode, Encode};
+use emulated_integration_tests_common::xcm_emulator::ConvertLocation;
+use frame_support::pallet_prelude::TypeInfo;
+use hex_literal::hex;
+use kusama_system_emulated_network::BridgeHubKusamaParaSender as BridgeHubKusamaSender;
+use snowbridge_beacon_primitives::CompactExecutionHeader;
+use snowbridge_core::{
+	inbound::{Log, Message, Proof},
+	outbound::OperatingMode,
+};
+use snowbridge_pallet_inbound_queue_fixtures::{
+	register_token_with_insufficient_fee::make_register_token_with_infufficient_fee_message,
+	InboundQueueFixture,
+};
+use snowbridge_router_primitives::inbound::{
+	Command, Destination, GlobalConsensusEthereumConvertsFor, MessageV1, VersionedMessage,
+};
+use sp_core::H256;
+use sp_runtime::{DispatchError::Token, TokenError::FundsUnavailable};
+use system_parachains_constants::kusama::snowbridge::EthereumNetwork;
+
+const INITIAL_FUND: u128 = 5_000_000_000 * KUSAMA_ED;
+const CHAIN_ID: u64 = 1;
+const TREASURY_ACCOUNT: [u8; 32] =
+	hex!("6d6f646c70792f74727372790000000000000000000000000000000000000000");
+const WETH: [u8; 20] = hex!("87d1f7fdfEe7f651FaBc8bFCB6E086C278b77A7d");
+const ETHEREUM_DESTINATION_ADDRESS: [u8; 20] = hex!("44a57ee2f2FCcb85FDa2B0B18EBD0D8D2333700e");
+
+#[derive(Encode, Decode, Debug, PartialEq, Eq, Clone, TypeInfo)]
+pub enum ControlCall {
+	#[codec(index = 3)]
+	CreateAgent,
+	#[codec(index = 4)]
+	CreateChannel { mode: OperatingMode },
+}
+
+#[allow(clippy::large_enum_variant)]
+#[derive(Encode, Decode, Debug, PartialEq, Eq, Clone, TypeInfo)]
+pub enum SnowbridgeControl {
+	#[codec(index = 83)]
+	Control(ControlCall),
+}
+
+pub fn send_inbound_message(fixture: InboundQueueFixture) -> DispatchResult {
+	EthereumBeaconClient::store_execution_header(
+		fixture.message.proof.block_hash,
+		fixture.execution_header,
+		0,
+		H256::default(),
+	);
+
+	EthereumInboundQueue::submit(
+		RuntimeOrigin::signed(BridgeHubKusamaSender::get()),
+		fixture.message,
+	)
+}
+
+/// Create an agent on Ethereum. An agent is a representation of an entity in the Polkadot
+/// ecosystem (like a parachain) on Ethereum.
+#[test]
+#[ignore]
```

Comment:
@claravanstaden why `#[ignore]`? (also in `polkadot-sdk`) Is there anything missing? If so, I would remove them and create separate issue to `polkadot-sdk` and fix them

Reply:
Create agent and create channel is disabled for initial launch and just removed from the safe call filter in the runtimes. If you prefer I can remove these tests and add them again when we enable these operations.

Reply:
> ...when we enable these operations.

When this will happen? Does it depend on anything event, release or what?

I mean, "remove now and re-add when enabling those operations" is ok, but also leave them here is ok, but in that case, please add there some more info "why ignored"  + TODO-issue e.g. `#[ignore = "TODO:issue#XYZ blahblah"]`

Reply:
I think it's more of a business decision, not a technical one since the operations are technically supported.